index.js

```diff
@@ -601,6 +601,17 @@ class Analytics {
     }
 
     if (!this.queue.length) {
+      if (this.pendingFlush) {
+        this.logger.debug('queue is empty, but a flush already exists');
+        // We attach the callback to the end of the chain to support a caller calling `flush()` multiple times when the queue is empty.
+        this.pendingFlush = this.pendingFlush
+          .then(() => {
+            callback();
+            return Promise.resolve();
+          });
```

Comment:
```suggestion
          .then((data) => {
            callback();
            return Promise.resolve(data);
          });
```

Reply:
I don't think this should be done. When we get to this code, the queue is empty, which means the callback for this call should resolve with no data. The only reason we're making the callback wait for the pending flush is so clients who are relying on flush() returning to shut down can also wait for pending flushes.

I can make this change, but it would mean that two callbacks to flush() might contain the same data. This might break certain cases. Let me know what you think.

Reply:
Let's do it without resolving data.